psa_crypto variants of drivers/entropy/api and crypto/rand32 tests fail to build for nrf9160dk_nrf9160_ns and nrf5340dk_nrf5340_cpuapp_ns

**Describe the bug**
The following tests:
- `drivers.rand32.random_psa_crypto` (in tests/crypto/rand32/)
- `drivers.entropy.psa_crypto` (in tests/drivers/entropy/api/)

cannot be built for the following boards:
- `nrf9160dk_nrf9160_ns`
- `nrf5340dk_nrf5340_cpuapp_ns`

This happens because the dts overlays used for those tests add "zephyr,psa-crypto-rng" nodes while such nodes (but with different names) already exists for SoCs used by the above boards and the related driver supports only single instance.

**To Reproduce**
Steps to reproduce the behavior:
1. `west twister -p nrf9160dk_nrf9160_ns -s tests/crypto/rand32/drivers.rand32.random_psa_crypto`

**Logs and console output**
```
/home/andy/zephyrproject/zephyr-sdk-0.15.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/subsys/random/libsubsys__random.a(rand32_entropy_device.c.obj): in function `z_impl_entropy_get_entropy':
/home/andy/zephyrproject/zephyr/include/zephyr/drivers/entropy.h:79: undefined reference to `__device_dts_ord_6'
/home/andy/zephyrproject/zephyr-sdk-0.15.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/subsys/random/libsubsys__random.a(rand32_entropy_device.c.obj): in function `rand_get':
/home/andy/zephyrproject/zephyr/subsys/random/rand32_entropy_device.c:77: undefined reference to `__device_dts_ord_6'
/home/andy/zephyrproject/zephyr-sdk-0.15.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/subsys/random/libsubsys__random.a(rand32_entropy_device.c.obj): in function `z_impl_sys_rand32_get':
/home/andy/zephyrproject/zephyr/subsys/random/rand32_entropy_device.c:37: undefined reference to `__device_dts_ord_6'
/home/andy/zephyrproject/zephyr-sdk-0.15.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/kernel/libkernel.a(init.c.obj): in function `z_impl_entropy_get_entropy':
/home/andy/zephyrproject/zephyr/include/zephyr/drivers/entropy.h:79: undefined reference to `__device_dts_ord_6'
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.15.1
 - Commit bec12a098d14172c53a090d84300d0e236357d56

--- a/tests/crypto/rand32/entropy_psa_crypto.overlay
+++ b/tests/crypto/rand32/entropy_psa_crypto.overlay
@@ -6,10 +6,10 @@
 
 / {
 	chosen {
-		zephyr,entropy = &rng_psa;
+		zephyr,entropy = &psa_rng;
 	};
 
-	rng_psa: entropy_psa_crypto {
+	psa_rng: psa-rng {
 		compatible = "zephyr,psa-crypto-rng";
 		status = "okay";
 	};

--- a/tests/drivers/entropy/api/entropy_psa_crypto.overlay
+++ b/tests/drivers/entropy/api/entropy_psa_crypto.overlay
@@ -6,10 +6,10 @@
 
 / {
 	chosen {
-		zephyr,entropy = &rng_psa;
+		zephyr,entropy = &psa_rng;
 	};
 
-	rng_psa: entropy_psa_crypto {
+	psa_rng: psa-rng {
 		compatible = "zephyr,psa-crypto-rng";
 		status = "okay";
 	};

